lib/iris/tests/unit/analysis/stats/test_pearsonr.py

```diff
@@ -35,18 +40,34 @@ def setUp(self):
 
 
 @tests.skip_data
-class TestLazy(Mixin, tests.IrisTest):
-    def test_perfect_corr(self):
+class TestLazy(Mixin):
+    @pytest.fixture
+    def mocked_compute(self, monkeypatch):
+        m_compute = mock.Mock(wraps=dask.base.compute)
+
+        # The three dask compute functions are all the same function but monkeypatch
+        # does not automatically know that.
+        # https://stackoverflow.com/questions/77820437
+        monkeypatch.setattr(dask.base, dask.base.compute.__name__, m_compute)
+        monkeypatch.setattr(dask, dask.compute.__name__, m_compute)
+        monkeypatch.setattr(dask.array, dask.array.compute.__name__, m_compute)
+
+        return m_compute
```

Comment:
Possibly we might want this in a `conftest.py` somewhere so it's available to other tests?

Reply:
Like maybe this one 👀 https://github.com/SciTools/iris/blob/main/lib/iris/tests/unit/conftest.py

Reply:
ping @ESadek-MO, who is planning some team discussions on this sort of thing soon.